src/main/java/cart/infrastructure/ProductDao.java

```diff
@@ -0,0 +1,88 @@
+package cart.infrastructure;
+
+import cart.domain.product.Product;
+import lombok.RequiredArgsConstructor;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Repository
+public class ProductDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public Product insert(Product product) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        jdbcTemplate.update(productPreparedStatementCreator(product), keyHolder);
+
+        return Product.builder()
+                .id(keyHolder.getKeyAs(Long.class))
+                .name(product.getName())
+                .imageUrl(product.getImageUrl())
+                .price(product.getPrice())
+                .build();
+    }
+
+    public Optional<Product> findById(Long id) {
+        String sql = "SELECT * FROM product WHERE ID = ?";
+
+        Optional<Product> product;
+        try {
+            product = Optional.ofNullable(jdbcTemplate.queryForObject(sql, productRowMapper(), id));
+        } catch (EmptyResultDataAccessException exception) {
+            product = Optional.empty();
+        }
+        return product;
```

Comment:
변수를 할당하지 않고 바로 리턴해도 될 것 같아요!

```suggestion
        try {
            return Optional.ofNullable(jdbcTemplate.queryForObject(sql, productRowMapper(), id));
        } catch (EmptyResultDataAccessException exception) {
            return Optional.empty();
        }
```

Reply:
앗 그러네요! 감사합니다. 개선해볼게요 :)